src/libstd/sys/wasi/fs.rs

```diff
@@ -616,14 +616,37 @@ fn open_at(fd: &WasiFd, path: &Path, opts: &OpenOptions) -> io::Result<File> {
     Ok(File { fd })
 }
 
+/// Get pre-opened file descriptors and their paths.
+fn get_paths() -> Result<Vec<(PathBuf, WasiFd)>, wasi::Error> {
+    use crate::sys::os_str::Buf;
+
+    let mut paths = Vec::new();
+    for fd in 3.. {
+        match wasi::fd_prestat_get(fd) {
+            Ok(wasi::Prestat { pr_type: wasi::PREOPENTYPE_DIR, u }) => unsafe {
+                let name_len = u.dir.pr_name_len;
+                let mut buf = vec![0; name_len];
+                wasi::fd_prestat_dir_name(fd, &mut buf)?;
+                let buf = Buf { inner: buf };
```

Comment:
Can this be replaced with `PathBuf::from(OsString::from_vec(buf))`?

Reply:
AFAIK there is no such method for [`OsString`](https://doc.rust-lang.org/src/std/ffi/os_str.rs.html).

Reply:
I see it's used in the file: https://github.com/rust-lang/rust/blob/804bbb2787f98fd264e82cf4868f82b6c244cc16/src/libstd/sys/wasi/fs.rs#L223

Probably comes from `use crate::os::wasi::ffi::{OsStrExt, OsStringExt};`.

Reply:
Yes, you are [right](https://github.com/rust-lang/rust/blob/master/src/libstd/sys_common/os_str_bytes.rs#L193).